### PR TITLE
ec2_scaling_policy boto3 rewrite for step scaling

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Convert ec2_scaling_policy to boto3 and implement step scaling

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_scaling_policy

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel e612d0be12) last updated 2017/07/05 16:42:21 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

##### ADDITIONAL INFORMATION
Tests require IAM privileges to create LaunchConfigs, AutoScalingGroups and Scaling Polices

### CODE REVIEW

**Comment by willthames** [comment]
WIP until #26233 is merged so that the IAM policies can be put in place in a modular fashion

**Comment by gundalow** [comment]
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

File: lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
 #!/usr/bin/python
-# Copyright: Ansible Project
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
-
-from __future__ import absolute_import, division, print_function
-__metaclass__ = type
-
+# This file is part of Ansible
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: Isn't the current copyright right?

File: test/integration/targets/ec2_scaling_policy/aliases [hunk 2]
```diff
@@ -0,0 +1,2 @@
+cloud/aws
+posix/ci/cloud/aws
```
Comment (s-hertel) [inline test/integration/targets/ec2_scaling_policy/aliases]: This alias needs to be updated to run in CI

File: lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py [hunk 3]
```diff
@@ -31,155 +41,353 @@
...
+        if module.params['estimated_instance_warmup']:
+            params['EstimatedInstanceWarmup'] = module.params['estimated_instance_warmup']
 
-        # check the remaining attributes
-        for attr in ('adjustment_type', 'scaling_adjustment', 'cooldown'):
-            if getattr(policy, attr) != module.params.get(attr):
+    try:
+        policies = connection.describe_policies(AutoScalingGroupName=asg_name,
+                                                PolicyNames=[policy_name])['ScalingPolicies']
+    except botocore.exceptions.ClientError as e:
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: BotoCoreError should be handled for the API calls as well.

File: lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py [hunk 4]
```diff
@@ -31,155 +41,353 @@
...
-    state = module.params.get('state')
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
 
     try:
-        connection = connect_to_aws(boto.ec2.autoscale, region, **aws_connect_params)
-    except (boto.exception.NoAuthHandlerFound, AnsibleAWSError) as e:
-        module.fail_json(msg=str(e))
+        connection = boto3_conn(module, conn_type='client', resource='autoscaling',
+                                region=region, endpoint=ec2_url, **aws_connect_params)
+    except (botocore.exceptions.NoCredentialsError, botocore.exceptions.ProfileNotFound) as e:
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: The connection doesn't need exception handling in the module. You could also update this to use AnsibleAWSModule and connection = module.client('autoscaling') if you wanted.

**Review by s-hertel — COMMENTED** [review]
A few trivial things, but looks great besides that.

File: lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py (hunk 4 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: Yeah, AnsibleAWSModule didn't exist when I made this PR :)

File: lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: Umm, it does look that way doesn't it. Again, an old change before the copyright standard was locked down, at least in my head!

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_scaling_policy.py]: Ah, I know what happened - the copyright and `from __future__` stuff got changed well after this PR was made in  9c81257c

**Comment by mkrizek** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.aws.

Please re-submit this issue in the above repository.

If you have further questions please stop by IRC or the mailing list:

* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**